### PR TITLE
Decontextualize multi-'yield' Parsing

### PR DESCRIPTION
**Pull request by CodaFi**
Yield statements support the ability to yield multiple values out of the coroutine context. These are modeled like an argument list in the grammar and parse sort of like one (minus labels). But this means there's a pretty serious grammatical ambiguity with applies

func yield() {}

yield ()

Rather than contextualize the parse, we're going to ignore the contextuality of multi-yields. The swift parser will always unambiguously parse 'yield &' as a yield statement, regardless of context. It will similarly unambiguously parse 'yield (' as an apply of a function called 'yield' regardless of context. We will then rewrite this AST on the way out to ASTGen where we actually have the ability to make context-sensitive decisions about the parse.

Fixes a regression introduced by #864 in the parsing of the macOS 11.3 copy of the standard library's swift interface.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci test

File: Sources/SwiftParser/Declarations.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 extension TokenConsumer {
   func atStartOfDeclaration(
-    context: Parser.ItemContext? = nil,
```
Comment (DougGregor) [inline Sources/SwiftParser/Declarations.swift:17]: I love this context going away!

File: Sources/SwiftParser/Statements.swift [hunk 2]
```diff
@@ -1106,10 +1106,23 @@ extension Parser.Lookahead {
...
+            return true
+          case .leftParen:
+            // "yield (", by contrast, must be disambiguated with additional
+            // context. We always consider it an apply expression of a function
+            // called `yield` for the purposes of the parse.
+            return false
+          default:
+            // "yield" followed by any other token is likely a yield statement
+            // of some singular expression.
+            return true
```
Comment (DougGregor) [inline Sources/SwiftParser/Statements.swift:1122]: Hmm. Should we check whether the next token is on a new line?

```swift
x = yield
print("huh")
```

?

**Review by DougGregor — APPROVED** [review]

File: Sources/SwiftParser/Statements.swift (hunk 2 above)
Comment (CodaFi) [reply Sources/SwiftParser/Statements.swift:1122]: Ooh, good catch.

Comment (CodaFi) [reply Sources/SwiftParser/Statements.swift]:
```
yield
print("huh")
```

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

**Pull request merged**